lib/steps/step_list.go

```diff
@@ -48,3 +55,34 @@ func (stepList *StepList) Prepend(step Step) {
 func (stepList *StepList) PrependList(otherList StepList) {
 	stepList.List = append(otherList.List, stepList.List...)
 }
+
+// WrapOptions represents the options given to Wrap.
+type WrapOptions struct {
+	RunInGitRoot     bool
+	StashOpenChanges bool
+}
+
+// Wrap wraps the given StepList in steps that
+// change to the Git root directory or stash away open changes.
+func (stepList *StepList) Wrap(options WrapOptions) {
```

Comment:
Let's take this idea all the way to the end:
- We always need to wrap steps, so wrapping steps is a built-in functionality of the StepList. It always does it.
- the `List` field is made private - nobody except the StepList itself should modify it
- rename the `Wrap` function to `Steps`: we always want the list of steps, properly wrapped
- how `Steps` wraps the result is parameterized when creating the StepList

```go
type StepList struct {
	list []Step
	StashOpenChanges bool
	RunInGitRoot bool
}

func (stepList *StepList) Steps() []Step {
  // this is the "Wrap" function, renamed to "Steps"
}


// using the StepList
stepList = steps.StepList{StashOpenChanges: true, RunInGitRoot: false}
// add steps
steps = StepList.Steps()
```

If we don't want to wrap, for example for the abort or undo cases, we simply provide `StashOpenChanges: false`.
What do you think about this?